pypdf/_writer.py

```diff
@@ -2618,13 +2618,16 @@ def merge(
                     cast(DictionaryObject, self._root_object["/AcroForm"])["/Fields"],
                 )
             trslat = self._id_translated[id(reader)]
-            for f in reader.trailer["/Root"]["/AcroForm"]["/Fields"]:  # type: ignore
-                try:
-                    ind = IndirectObject(trslat[f.idnum], 0, self)
-                    if ind not in arr:
-                        arr.append(ind)
-                except KeyError:
-                    pass
+            try:
+                for f in reader.trailer["/Root"]["/AcroForm"]["/Fields"]:  # type: ignore
+                    try:
+                        ind = IndirectObject(trslat[f.idnum], 0, self)
+                        if ind not in arr:
+                            arr.append(ind)
+                    except KeyError:
+                        pass
+            except KeyError:
+                arr = self._add_object(ArrayObject())
```

Comment:
Would adding an empty array with the key `"/Fields"` to `"/AcroForm"` have the same effect? Would it cause other issues?

Alternatively, what do you think about this?

```suggestion
            if "/Fields" not in reader.trailer["/Root"]["/AcroForm"]:
                arr = self._add_object(ArrayObject())
            else:
                for f in reader.trailer["/Root"]["/AcroForm"]["/Fields"]:  # type: ignore
                    try:
                        ind = IndirectObject(trslat[f.idnum], 0, self)
                        if ind not in arr:
                            arr.append(ind)
                    except KeyError:
                        pass
```

Reply:
From my understanding an Empty Array is the default value (no fields in the document)

Reply:
I've aded a commit to remove unused code (based on Codecov report)